MetadataExtractor/Formats/Bmp/BmpReader.cs

```diff
@@ -347,8 +347,12 @@ private static void ReadBitmapHeader(SequentialReader reader, BmpHeaderDirectory
                         }
                         else
                         {
-                            var iccReader = new ByteArrayReader(reader.GetBytes(profileSize));
-                            var iccDirectory = new IccReader().Extract(iccReader);
+                            var iccBytes = profileSize <= 256
+                                ? stackalloc byte[profileSize]
+                                : new byte[profileSize];
```

Comment:
I would love to wrap this pattern up somehow, to:

- Have the constant in a single place
- Make it easy to use a pooled array when over the threshold

I haven't looked at this yet. We can file an issue for later.

Reply:
Will tackle in this PR. Will look something like:

```
byte[]? rentedBuffer = null;

var iccBytes = profileSize <= 256
    ? stackalloc byte[profileSize]
    : (rentedBuffer = ArrayPool.Shared.Rent(profileSize)).Slice(0, profileSize);

...

if (rentedBuffer != null)
{
    ArrayPool.Shared.Return(rentedBuffer);
}

```

Reply:
I'm wondering if we can make the call site look like this:

```c#
using rental = BufferUtil.Rent(minimumSize: 1234);

var reader = new BufferReader(rental.Buffer);
```

I'm not familiar enough with `ref struct`/`scoped` here to know if this is possible but I think it is. I think we can also return a `struct` disposable and the compiler won't box it to invoke `Dispose` through the interface virtually.

If we had a pattern like that, we could look at making the stack allocation smarter. I read something somewhere about being able to test for stack space, which might mean the threshold could be dynamic. Even a threshold of 256 could blow the stack. However I don't know if such a capability exists, or if I just read a proposal for that.